Angular/Server-side rendering (SSR)/Robots and sitemap.md

```diff
@@ -0,0 +1,114 @@
+
+You mark which search engine you want to address with the name attribute, and the content attribute indicates the desired action. In this example, the tag prevents the content from being indexed by all search engines.
+
+We will crete function for setting robots depending on the environment type:
+
+```typescript
+export function initIndexing(
+	environmentVariablesService: EnvironmentVariablesService,
+	meta: Meta
+): () => Promise<void> {
+	return async () => {
+		const isProduction = environmentVariablesService.get(EnvironmentVariable.APP_ENVIRONMENT) === 'production';
+
+		meta.removeTag('name="robots"');
+		meta.addTag({
+			name: 'robots',
+			content: isProduction ? 'index, follow' : 'noindex, nofollow',
+		});
+	};
+}
+```
+
+And add this on app initiation in the `app.module.ts`:
+
+```typescript
+@NgModule({
+    providers: [
+        {
+            provide: APP_INITIALIZER,
+            useFactory: initIndexing,
+            multi: true,
+            deps: [EnvironmentVariablesService, Meta],
+        },
+    ]
+})
+```
+
+### What is robots.txt?
+A robots.txt file (Robots Exclusion Standard Protocol) is a text file that tells search engine crawlers which files or pages they can crawl.
+
+The search engine or its crawler is identified in the robots.txt file with user-agent. To disallow and allow commands can be used to specify which directories should and should not be crawled. You can also refer to the location of a sitemap in the robots.txt file: 
+
+```text
+User-agent: *
+Disallow: /
+```
+
+We can use express to return content for the robots.txt file. In the `server.ts` add this:
+
+```typescript
+const nonIndexableRobotsContent = 'User-agent: *\nDisallow: /';
+const indexableRobotsContent = 'User-agent: *\nDisallow:';
+const isProduction = process.env[EnvironmentVariable.APP_ENVIRONMENT] === 'production';
+
+server.get('/robots.txt', (req, res) => {
+		res.type('text/plain');
+		res.send(isProduction ? indexableRobotsContent : nonIndexableRobotsContent);
+	});
+```
+
+## Sitemap
+An XML sitemap is a file that lists a website’s important pages, making sure Google can find and crawl them all. It also helps search engines understand your website structure. You want Google to crawl every essential page of your website. But sometimes, pages end up without any internal links pointing to them, making them hard to find. A sitemap can help speed up content discovery.
+
+To return sitemap we will also use express for this and return custom content. For this to work we will need to install these libraries:
+
+ * [apicache](https://www.npmjs.com/package/apicache) for caching the response
+ * [sitemap](https://www.npmjs.com/package/sitemap) for generating sitemap.xml file
+
+```typescript
+server.get('/sitemap.xml', cache('1 hour'), (req, res) => {
+		res.header('Content-Type', 'application/xml');
+
+		// If there is no API_URL environment variable return server error and stop creating sitemap
+		if (!apiUrl) {
```

Comment:
I will remove this part as we had translations in both projects, so it can be a bit confusing for other people where route translations are used. We could add route translation as a separate chapter in the handbook for Angular.